src/pixmap.c

```diff
@@ -300,6 +301,70 @@ void pixmap_free(struct pixmap* pm)
     free(pm->data);
 }
 
+bool pixmap_save(struct pixmap* pm, const char* path)
+{
+    FILE* fp;
+    unsigned i;
+
+    if (!(fp = fopen(path, "wb"))) {
+        return false;
+    }
+
+    // TODO: add alpha channel
```

Comment:
We have 14 supported image formats. It would be better to use one of them instead of reinventing own format.

Reply:
I don't see any function for writing an image to disk in the codebase. Did I miss something?

Reply:
Current code doesn't contain code for encoders. But these APIs are available to `swayimg`.

Reply:
I think it would be better to add `encode_pnm` to [PNM(PPM)](https://github.com/artemsen/swayimg/blob/master/src/formats/pnm.c) decoder.